doc/sphinx/conf.py

```diff
@@ -42,8 +42,118 @@
               'sphinxarg.ext',
               'sphinxcontrib.doxylink',
               'sphinx.ext.intersphinx',
+              'sphinx_gallery.gen_gallery',
+              'sphinx_tags',
+              'sphinx_design',
               ]
 
+sphinx_gallery_conf = {
+    'filename_pattern': '\.py',
+    'example_extensions': {'.py', '.cpp', '.h', '.c', '.f', '.f90', '.m'},
+    "filetype_parsers": {'.h': 'C++', '.m': 'Matlab'},
+    'ignore_pattern': r'(__.*__\.py|test_examples\.m)',
+    'image_srcset': ["2x"],
+    'examples_dirs': [
+       '../samples/python/',
+       '../samples/cxx/',
+       '../samples/clib/',
+       '../samples/fortran/',
+       '../samples/matlab_experimental/',
+    ],
+    'gallery_dirs': [
+       'examples/python',
+       'examples/cxx',
+       'examples/clib',
+       'examples/fortran',
+       'examples/matlab_experimental',
+    ],
+    'reference_url': {
+        'cantera': None,  # 'None' means the locally-documented module
+    }
+}
+
+# Override sphinx-gallery's method for determining which examples should be executed.
+# There's really no way to achieve this with the `filename_pattern` option, and
+# `ignore_pattern` excludes the example entirely.
+skip_run = {
+    # multiprocessing can't see functions defined in __main__ when run by
+    # sphinx-gallery, at least on macOS.
+    "multiprocessing_viscosity.py",
+    # __file__ deliberately not available when run by sphinx-gallery
```

Comment:
https://sphinx-gallery.github.io/stable/faq.html#why-is-file-not-defined-what-can-i-use

Reply:
Ah, ok. To me, this wasn’t clear. I guess we could avoid this by using `os.getcwd()` as suggested?

Reply:
Maybe, but this isn't exactly the same. For example, when this is run as part of the `run-examples` workflow, the working directory is the root of the repository. The output for this example isn't that interesting anyway -- it looks pretty much just like the other simple flame examples, so I don't think we're missing much.

Reply:
True. At the same time, I hate to see an example that some users may find interesting left behind. I’d suggest to add the contents of `tdata.dat` as a multi-line string within the Python script, instead of omitting it.

Reply:
Ah, to be clear: the example still appears in the documentation. We just don't run it and capture output (so, same as what's happening for the non-Python examples). However, I realize we do need to provide a link to the `tdata.dat` though, for anyone who downloads the example.

Reply:
👍 makes sense. At the same time, we recently had a confused user asking about the format to be used for this exact example. I believe that including data in the file itself would resolve all issues.